src/Network/Algorand/Node/Api.hs

```diff
@@ -62,6 +65,49 @@ data Version = Version
   deriving (Generic, Show)
 $(deriveJSON algorandSnakeOptions 'Version)
 
+newtype NanoSec = NanoSec { unNanoSec :: Integer }
```

Comment:
My hatred towards unbounded types is probably not well-justified, but may I suggest using `Int64` here instead of `Integer` (since this is what the Go implementation uses)?

Reply:
Changed to `Word64` as in the daemon for now, though that's probably not good for subtraction.